common/src/main/java/software/tnb/common/utils/waiter/Waiter.java

```diff
@@ -0,0 +1,127 @@
+package software.tnb.common.utils.waiter;
+
+import software.tnb.common.exception.FailureConditionMetException;
+import software.tnb.common.exception.TimeoutException;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+/**
+ * Class to define the waiting conditions for the {@link software.tnb.common.utils.WaitUtils#waitFor(Waiter)} method.
+ */
+public class Waiter {
+    private final BooleanSupplier condition;
+    private final String logMessage;
+
+    private BooleanSupplier failureCondition;
+    private int retries = 24;
+    private long retryTimeout = 5000L;
+    private String failureMessage = "Specified fail condition met";
+
+    private Supplier<FailureConditionMetException> failureException = () -> new FailureConditionMetException(failureMessage);
+    private Supplier<TimeoutException> timeoutException = () -> new TimeoutException("Timeout exceeded");
+
+    /**
+     * Constructor.
+     * @param condition success condition
+     * @param logMessage log message printed at the start of the wait
+     */
+    public Waiter(BooleanSupplier condition, String logMessage) {
```

Comment:
Don't you think that a builder approach is a little bit useful? I mean a static method to instantiate the class, instead of the constructor?

Reply:
these two (`condition` and `logMessage`) are always required, that's why I chose to use the constructor for them, all others that are exposed through the builder are optional